modules/airavata-helix/task-core/src/main/java/org/apache/airavata/helix/core/participant/HelixParticipant.java

```diff
@@ -41,27 +41,29 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 /**
  * TODO: Class level comments please
  *
  * @author dimuthu
  * @since 1.0.0-SNAPSHOT
  */
-public class HelixParticipant <T extends AbstractTask> implements Runnable {
+public class HelixParticipant<T extends AbstractTask> implements Runnable {
 
     private final static Logger logger = LoggerFactory.getLogger(HelixParticipant.class);
 
+    private final static int PARTICIPANT_SHUTDOWN_GRACE_PERIOD = 30000;
```

Comment:
Make this externally configurable using setter method and 3000 is the default value if it is not set externally.